### PR TITLE
[BACKLOG-371101] GWTTree**.java - Functional images don't provide tex…

### PR DESCRIPTION
**Pull request by NJtwentyone**
…t alternative - GWT based Images

 - update modeler.xul to contain new <treecol> attribute "pen:alttextbinding"
 - added new properties to modeler.properties
 
 1/3 Pull requests:

- https://github.com/pentaho/modeler/pull/376
- https://github.com/pentaho/data-access/pull/1194
- https://github.com/pentaho/pentaho-commons-xul/pull/226

see demo and explanation at: https://hv-eng.atlassian.net/browse/BACKLOG-37101?focusedCommentId=1743972

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 48m 12s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/data-access-plugin/job/PR-1194/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties [hunk 1]
```diff
@@ -142,6 +142,14 @@ validation.level.MISSING_LAT=Level must have a Latitude member property
 validation.level.MISSING_LON=Level must have a Longitude member property
 default_aggregation=Default Aggregation
 
+# Model Tree alternative text
+modeler.alternative_text.model=Model
```
Comment (NJtwentyone) [inline core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties:146]: did not add the related properties for the internationalization files for English, Japanese, etc.. (ex:  modeler_en.properties, modeler_jp.properties). 

I was told PM will make this decision at the end of release and an magic process will happen

File: core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties [hunk 2]
```diff
@@ -142,6 +142,14 @@ validation.level.MISSING_LAT=Level must have a Latitude member property
...
+modeler.alternative_text.dimension=Dimension
+modeler.alternative_text.category=Category
+modeler.alternative_text.baseColumnBacked=Column
+modeler.alternative_text.availableField=Column
```
Comment (NJtwentyone) [inline core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties:150]: These properties are consumed by the files in pentaho/modeler source code path : core/src/main/java/org/pentaho/agilebi/modeler/nodes/DimensionMetaData.java

see PR https://github.com/pentaho/modeler/pull/376 for such classes

File: core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties [hunk 3]
```diff
@@ -142,6 +142,14 @@ validation.level.MISSING_LAT=Level must have a Latitude member property
 validation.level.MISSING_LON=Level must have a Longitude member property
 default_aggregation=Default Aggregation
 
+# Model Tree alternative text
+modeler.alternative_text.model=Model
+modeler.alternative_text.dimension=Dimension
+modeler.alternative_text.category=Category
```
Comment (NJtwentyone) [inline core/src/main/resources-filtered/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.properties:148]: for deciding the text, based it off what the image looked like and/or the column it was in the data source modeler screen.

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.xul [hunk 4]
```diff
@@ -33,7 +33,8 @@
             <treecol flex="1" label="" primary="true"
                      pen:binding="name" pen:childrenbinding="children"
                      pen:imagebinding="image" editable="false"
-                     pen:classnamebinding="classname" pen:expandedbinding="expanded"/>
+                     pen:classnamebinding="classname" pen:expandedbinding="expanded"
+                     pen:alttextbinding="altText"/>
```
Comment (NJtwentyone) [inline core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/public/modeler.xul:37]: this is the glue code that integrates
GwtTreeCole#register
GwtTree#addTreeChild

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 8m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **359**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/data-access-plugin/job/PR-1194/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

**Review by ddiroma — APPROVED** [review]

**Pull request merged**